accepted/2023/simplify-workload-versioning-implementation.md

```diff
@@ -0,0 +1,129 @@
+- For file-based installs, it will be stored in `dotnet\metadata\workloads\<Feature Band>\InstallState`
+
+If the workload versions are "pinned", there will be a `default.json` file in this folder.  The file will specify either a workload set version, or versions for each individual manifest:
+
+```json
+{
+  "workloadVersion": "8.0.102"
+}
+```
+
+```json
+{
+  "manifests":
+  {
+    "microsoft.net.sdk.android": "33.0.46/7.0.100",
+    "microsoft.net.sdk.ios": "16.4.7054/7.0.100",
+    //  etc.
+  }
+}
+```
+
+Normally the file should use a workload set version if possible, and only include manifest versions when a rollback file is used.  However, if there are manifests that are installed that are not part of the workload set (for example for Tizen), then the json file should include the workload set version and then the manifest versions for manifests that are not part of the workload set.
+
+When workload versions are pinned, running `dotnet workload install` will not automatically do a workload update.
+
+If `dotnet workload update` is run, then any pinned workload versions will be removed (the json file will be deleted), as the intent of the command is to update to the latest version.
+
+TODO: Error if global.json specifies a workload set that isn't installed should recommend running `dotnet workload restore`.
+
+### Workload resolver updates
+
+Because the `workloadsets` folder will be next to the other workload manifest folders, the resolver will be updated to ignore this folder when looking for workload manifests.  Older versions of the resolver should not be impacted, because they will not be looking in the newer version band folder.  This does mean that the resolver changes should go into Visual Studio as soon as possible so that full Framework MSBuild can work with the new manifests.
+
+The workload resolver should first look for a workload set version from a global.json file.  If not specified via global.json, then it should look in the workload install state folder for a workload set or workload manifest versions to use.  If not specified via the install state folder, then the resolver should use the latest workload set version installed for the feature band.  Finally, if no workload set is installed, the resolver should use the latest version of each manifest for the feature band.
+
+### Installing workload set from global.json
+
+`dotnet workload update` - If workload set is specified in global.json, this should install that workload set.  Also should register this global.json path in workload install state folder as a GC root.
+
+- `dotnet workload restore` - global.json aware, same as update
+- `dotnet workload install` - global.json aware
+- `dotnet workload clean` - should look through all records of global.json, see if they are still there and up-to-date, delete as necessary before GC
+- `dotnet workload roots` - (not sure about command name) List all registered global.json files and versions of workload sets they pin
```

Comment:
Does this mean in some future, it's possible to have old workload packs around on your system. And then you'd have to delete a `global.json` to successfully remove those packs?

That seems OK: as the behavior now is you'd lose old packs that some project might be using.